### PR TITLE
Add Windows CI configuration (Travis CI)

### PR DESCRIPTION
**Pull request by lassik**
Copied from unibeautify core

### CODE REVIEW

**Comment by Glavin001** [comment]
🤞Building now: https://ci.appveyor.com/project/Glavin001/unibeautify-cli/builds/19435055

Thanks for making this PR, @lassik !

**Comment by lassik** [comment]
The build/test machinery now works, but some of the tests are failing for reasons I don't understand.

**Comment by stevenzeck** [comment]
Yeah, you'll want to follow the `.travis.yml` file in this repo.  Add a line to install `@unibeautify/beautifier-eslint` and `eslint@4`.  e.g. `npm install -g unibeautify @unibeautify/beautifier-eslint eslint@4`

**Comment by lassik** [comment]
Windows issues uncovered by failing tests.

Relative pathname turns into absolute pathname:

```
    Expected: "ENOENT: no such file or directory, open 'test/test2.js'"
    Received: "ENOENT: no such file or directory, open 'C:\\projects\\unibeautify-cli\\test\\test2.js'"
```

Newline issue in test fixture (Unix `\n` vs Windows `\r\n`):

```
    -   "stdout": "Supported languages
    +   "stdout": "Supported languages
```

Another:

```
    -   "stdout": "Supported beautifiers
    +   "stdout": "Supported beautifiers
```

**Comment by lassik** [comment]
Wait, I don't think those are newline issues after all. I guess the issue is that ANSI color doesn't work on Windows (without the third-party `ansi.sys` hack).

**Comment by lassik** [comment]
We seem to be using [chalk](https://github.com/chalk/chalk) for ANSI color like this.

    this.writeOut(chalk.blue(`Supported ${section}`));

Its [Windows support](https://github.com/chalk/chalk/search?q=windows) is... probably incomplete. I get the impression they rely on ANSI colors even on Windows. ANSI colors are not natively supported there. Instead, there are special WinAPI calls for terminal color. You don't write out a magic string, you call a function in a DLL.

**Comment by stevenzeck** [comment]
Maybe we can strip the ANSI colors from the results in the test script and update the snapshots, such as here:  https://github.com/Unibeautify/unibeautify-cli/blob/master/test/commands/SupportCommand.spec.ts#L39

**Comment by lassik** [comment]
Hmm. The CLI should really be adaptive so that it doesn't try to use color if it's not outputting to a terminal. The [Node TTY API](https://nodejs.org/api/tty.html) has `writeStream.isTTY` and `tty.isatty(fd)`. I guess the test framework makes the CLI output to a pipe or file instead of a pty (pseudo-TTY). If so, we could generate new test fixtures without the ANSI color codesand that would make the tests pass.

**Comment by lassik** [comment]
`writeOut` just calls `this.stdout.write()`. Is `this.stdout` a Node `writeStream`? Then we could just check for `this.stdout.isTTY`.

**Comment by stevenzeck** [comment]
Let's look into this as well as a potential replacement:  https://blog.travis-ci.com/2018-10-11-windows-early-release?utm_source=intercom&amp;utm_medium=web&amp;utm_campaign=windows_early_release

**Comment by lassik** [comment]
That's awesome! 🎉 And literally announced yesterday :D Would be great to be able to cover all platforms with the same CI service.

**Comment by Glavin001** [comment]
I'd prefer to use Travis CI for all OS support checks. 🎉

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+environment:
```
Comment (Glavin001) [inline appveyor.yml]: I'd prefer to remove AppVeyor and use Travis CI. 🎉

**Comment by lassik** [comment]
I'll wait for #101 to be merged before continuing with this. This PR is quite a mess, should we continue with it and squash-merge or start a new Travis Windows PR?

**Comment by stevenzeck** [comment]
@lassik could you move this into #101 instead?  Basically delete the appveyor config file and add Windows to Travis.

**Comment by lassik** [comment]
Yeah, that sounds like a better idea.

**Pull request closed without merging**